typescript-view-swapping/main.ts

```diff
@@ -0,0 +1,25 @@
+const $tabContainer = document.querySelector('.tab-container');
+if (!$tabContainer) throw new Error('$tabContainer not found!');
+$tabContainer.addEventListener('click', clickHandler);
+
+function clickHandler(event: Event): void {
+  const eventTarget = event.target as HTMLElement;
+  if (eventTarget.matches('.tab')) {
+    for (const tab of document.querySelectorAll('.tab')) {
+      if (tab === eventTarget) {
```

Comment:
We would want to query for these at the top of our `ts` file and store it in a variable. If we wait to query it inside of the event handler callback function, it would slow down our application because 1) we have to wait for the query here instead of when the page loads, and 2) we have to query _every_ time we click on a tab which means we're repeating work.